common/lib/xmodule/xmodule/modulestore/search.py

```diff
@@ -104,18 +105,31 @@ def find_path_to_course():
         # (e.g. sequential and videosequence) currently deal with this form of
         # representing nested positions. This needs to happen before jumping to a
         # module nested in more than one positional module will work.
+
         if n > 3:
             position_list = []
+            masquerading_as_student = False
+            if request and request.session.get('masquerade_settings') and \
+                request.session.get('masquerade_settings').get(course_id) and \
```

Comment:
we can make an internal method to check all these `conditions`. It will make readability better.